### PR TITLE
tests: uart: esp32: Test configuration

### PR DESCRIPTION
**Pull request by raffarost**
Add file configs for ESP32 devices for CI testing.
Get baudrate from device tree for `uart_basic_api` and `uart_elementary` tests.
Fix esp32 uart driver function `uart_get_config` to return standard baudrate value.

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

**Review by wmrsouza — DISMISSED** [review]

File: tests/drivers/uart/uart_elementary/socs/esp32s3_procpu.conf [hunk 1]
```diff
@@ -0,0 +1 @@
+CONFIG_DUAL_UART_TEST=y
```
Comment (sylvioalves) [inline tests/drivers/uart/uart_elementary/socs/esp32s3_procpu.conf:1]: I see that `ESP32-S2` and all `riscv` SoCs don´t have `CONFIG_DUAL_UART_TEST=y` set. Is that correct?

**Review by sylvioalves — DISMISSED** [review]

**Review by marekmatej — DISMISSED** [review]

**Review by sylvioalves — DISMISSED** [review]

**Review by wmrsouza — DISMISSED** [review]

File: tests/drivers/uart/uart_basic_api/src/test_uart_config.c [hunk 2]
```diff
@@ -25,13 +25,13 @@
...
-		.parity = UART_CFG_PARITY_NONE,
-		.stop_bits = UART_CFG_STOP_BITS_1,
-		.data_bits = UART_CFG_DATA_BITS_8,
-		.flow_ctrl = UART_CFG_FLOW_CTRL_NONE
-	};
+	.baudrate = DT_PROP_OR(DT_CHOSEN(zephyr_console), current_speed, 115200),
+	.parity = UART_CFG_PARITY_NONE,
+	.stop_bits = UART_CFG_STOP_BITS_1,
+	.data_bits = UART_CFG_DATA_BITS_8,
+	.flow_ctrl = UART_CFG_FLOW_CTRL_NONE};
```
Comment (dcpleung) [inline tests/drivers/uart/uart_basic_api/src/test_uart_config.c:34]: Please keep the closing bracket on another line.

File: tests/drivers/uart/uart_basic_api/src/test_uart_config_wide.c [hunk 3]
```diff
@@ -25,13 +25,13 @@
...
-		.parity = UART_CFG_PARITY_NONE,
-		.stop_bits = UART_CFG_STOP_BITS_1,
-		.data_bits = UART_CFG_DATA_BITS_9,
-		.flow_ctrl = UART_CFG_FLOW_CTRL_NONE
-	};
+	.baudrate = DT_PROP_OR(DT_NODELABEL(dut), current_speed, 115200),
+	.parity = UART_CFG_PARITY_NONE,
+	.stop_bits = UART_CFG_STOP_BITS_1,
+	.data_bits = UART_CFG_DATA_BITS_9,
+	.flow_ctrl = UART_CFG_FLOW_CTRL_NONE};
```
Comment (dcpleung) [inline tests/drivers/uart/uart_basic_api/src/test_uart_config_wide.c:34]: Ditto for closing bracket.

**Review by dcpleung — DISMISSED** [review]
Since you are amending the changes, please leave the whitespace changes out of functional changes. This will make future reviewing (and git log/show) of code easier to understand.

**Review by wmrsouza — DISMISSED** [review]

**Comment by raffarost** [comment]
hi @dcpleung, in the last commits I reverted the non-functional changes for the `elementary` test and added a Kconfig to allow configuring a value for the mismatch case. It could be fixed to 9600 as it was before, but since we are getting baudrate from the device tree, it made sense to let it be configurable as well.

**Review by sylvioalves — DISMISSED** [review]

File: tests/drivers/uart/uart_elementary/Kconfig [hunk 4]
```diff
@@ -8,4 +8,11 @@ config DUAL_UART_TEST
 config SETUP_MISMATCH_TEST
 	bool "Enable mismatched configuration in dual UART test"
 
+config UART_BAUDRATE_MISMATCH
+	int "UART baudrate for DUT_AUX to test mismatch case"
+	default 9600
+	help
+	  UART baudrate for DUT_AUX that is intentionally different from DUT.
+	  Used when SETUP_MISMATCH_TEST is enabled.
```
Comment (wmrsouza) [inline tests/drivers/uart/uart_elementary/Kconfig:16]: Do you mind making UART_BAUDRATE_MISMATCH explicitly dependent on SETUP_MISMATCH_TEST?

**Review by sylvioalves — DISMISSED** [review]

File: tests/drivers/uart/uart_elementary/Kconfig (hunk 4 above)
Comment (raffarost) [reply tests/drivers/uart/uart_elementary/Kconfig:16]: good idea

**Review by wmrsouza — DISMISSED** [review]

**Review by LucasTambor — DISMISSED** [review]

**Review by sylvioalves — APPROVED** [review]

**Review by wmrsouza — APPROVED** [review]

**Review by LucasTambor — APPROVED** [review]

**Review by marekmatej — APPROVED** [review]

**Comment by raffarost** [comment]
@dcpleung @nashif PTAL

**Comment by sylvioalves** [comment]
@dcpleung @kartben ping.

**Comment by kartben** [comment]
Co-assigning to @sylvioalves as platform owner and due to the nature of the changes. 
@dcpleung is on leave but I'm confident his previous NACK has been addressed.

**Review by kartben — APPROVED** [review]

**Pull request merged**